core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -348,7 +348,7 @@ protected final void onSendingBytes(int numBytes) {
    *
    * @param numBytes the number of bytes that were sent.
    */
-  protected final void onSentBytes(int numBytes) {
+  protected void onSentBytes(int numBytes) {
```

Comment:
I'm not seeing why you removed `final` here ... the overridden methods just call super.